src/components/common/Extensions/UserHub/partials/StakesTab/partials/StakeItem.tsx

```diff
@@ -1,33 +1,68 @@
-import React, { type FC } from 'react';
+import React, { useEffect, useState, type FC } from 'react';
 import { FormattedDate, useIntl } from 'react-intl';
-import { useNavigate } from 'react-router-dom';
+import { useNavigate, useParams } from 'react-router-dom';
 
 import { getActionTitleValues } from '~common/ColonyActions/index.ts';
+import { useGetFullColonyByAddressQuery } from '~gql';
 import { TX_SEARCH_PARAM } from '~routes';
 import Numeral from '~shared/Numeral/index.ts';
+import { setQueryParamOnUrl } from '~utils/urls.ts';
 import UserStakeStatusBadge from '~v5/common/Pills/UserStakeStatusBadge/index.ts';
 
 import { type StakeItemProps } from '../types.ts';
 
 const displayName =
   'common.Extensions.UserHub.partials.StakesTab.partials.StakeItem';
 
-const StakeItem: FC<StakeItemProps> = ({ nativeToken, stake, colony }) => {
+const StakeItem: FC<StakeItemProps> = ({ stake }) => {
   const { formatMessage } = useIntl();
   const navigate = useNavigate();
 
+  const [navigatePath, setNavigatePath] = useState(window.location.pathname);
+
   const stakeItemTitle =
     stake.action?.metadata?.customTitle ||
     stake.action?.decisionData?.title ||
     stake.action?.type;
 
+  const { colonyName: colonyNameUrl } = useParams();
+
+  const { data: stakeColonyData, loading: stakeColonyLoading } =
+    useGetFullColonyByAddressQuery({
+      variables: {
+        address: stake.action?.colonyAddress || '',
+      },
+      fetchPolicy: 'network-only',
+      nextFetchPolicy: 'cache-first',
+    });
```

Comment:
It's an expensive query to be making for each stake in the list, I think the best thing to do here would be to modify the `UserStake` model to include a reference to `colony` field. You can then extract just the native token and its decimals in one go.

You could go a step further if you wish and fetch the extensions in the UserStake fragment as well. The benefit of such approach is that it would also work for stakes in colonies that the user has subsequently left.